TradeBeacons/data/scripts/systems/tradingoverview.lua

```diff
@@ -102,28 +102,64 @@ function getTradeBeaconScanRange(seed, rarity)
 end
 
 function requestSectorsData(caller)
-    if not caller then
+    local player
+    if caller then
+        -- triggered by player interaction so we can be specific
+        player = Player(caller)
+    elseif onServer() then
+        local faction = getParentFaction()
+        if faction.isPlayer then
+            -- only one choice for the player
+            player = Player(faction.index)
+        elseif faction.isAlliance then
+            -- TODO: maybe we can pick one in a more principled way?
+            --  or maybe trade beacons should be registering themselves with
+            --  trading systems instead of players?
+
+            local alliance = Alliance(faction.index)
+            local players = {Server():getOnlinePlayers()}
+            for _, p in pairs(players) do
+                if alliance:contains(p.index) then
+                    player = p
+                    break
+                end
+            end
+            if not player then
+                -- no player is logged in, so we don't have anywhere to get the
+                -- data from, no-op
+                return
+            end
+        else
+            -- faction isn't a player or alliance, no-op
+            return
+        end
+    else
+        -- no caller and not on server, this is unexpected, so no-op
         return
     end
 
     local entityId = Entity().index.string
     local x, y = Sector():getCoordinates()
     local script = "tradebeacon.lua"
 
-    Player(caller):invokeFunction(script, "requestSectorsData", x, y, historySize, entityId, caller)
+    player:invokeFunction(script, "requestSectorsData", x, y, historySize, entityId, caller)
```

Comment:
note that caller can be nil now. the player script already will just pass this through since it only needs the entityId to be non-nil to do the invokeRemoteEntityFunction call